tensordict/nn/distributions/composite.py

```diff
@@ -147,19 +167,71 @@ def rsample(self, shape=None) -> TensorDictBase:
             shape + self.batch_shape,
         )
 
-    def log_prob(self, sample: TensorDictBase) -> TensorDictBase:
-        """Writes a ``<sample>_log_prob entry`` for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob."""
+    def log_prob(self, sample: TensorDictBase) -> torch.Tensor:
+        """Computes and returns the summed log-prob."""
+        if not self.aggregate_probabilities:
+            return self.log_prob_composite(sample, include_sum=True)
+        slp = 0.0
+        for name, dist in self.dists.items():
+            lp = dist.log_prob(sample.get(name))
+            while lp.ndim > sample.ndim:
+                lp = lp.sum(-1)
+            slp = slp + lp
+        return slp
+
+    def log_prob_composite(
+        self, sample: TensorDictBase, include_sum=True
+    ) -> TensorDictBase:
+        """Writes a ``<sample>_log_prob`` entry for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob."""
         slp = 0.0
         d = {}
         for name, dist in self.dists.items():
             d[_add_suffix(name, "_log_prob")] = lp = dist.log_prob(sample.get(name))
             while lp.ndim > sample.ndim:
                 lp = lp.sum(-1)
             slp = slp + lp
-        d[self.log_prob_key] = slp
+        if include_sum:
+            d[self.log_prob_key] = slp
         sample.update(d)
         return sample
 
+    def entropy(self, samples_mc=1) -> torch.Tensor:
+        """Computes and returns the summed entropies."""
+        if not self.aggregate_probabilities:
+            return self.entropy_composite(samples_mc, include_sum=True)
+        se = 0.0
+        for _, dist in self.dists.items():
+            try:
+                e = dist.entropy()
+            except NotImplementedError:
+                x = dist.rsample((samples_mc,))
+                e = -dist.log_prob(x).mean(0)
+            while e.ndim > len(self.batch_shape):
```

Comment:
this seems to lead to a different result than log-prob, which makes sense because we don't pass a TD. I think it should be documented though: each docstring should tell what shape of tensor output is to be expected.